src/hooks/useTxPendingStatuses.ts

```diff
@@ -32,17 +33,24 @@ const useTxMonitor = (): void => {
       return
     }
 
-    for (const [txId, { txHash, status }] of pendingTxEntriesOnChain) {
-      const isProcessing = status === PendingStatus.PROCESSING
+    for (const [txId, { txHash, status, taskId }] of pendingTxEntriesOnChain) {
+      const isProcessing = status === PendingStatus.PROCESSING && txHash !== undefined
       const isMonitored = monitoredTxs.current[txId]
+      const isRelaying = status === PendingStatus.RELAYING && taskId !== undefined
 
-      if (!txHash || !isProcessing || isMonitored) {
+      if (!(isProcessing || isRelaying) || isMonitored) {
         continue
       }
 
       monitoredTxs.current[txId] = true
 
-      waitForTx(provider, txId, txHash)
+      if (isProcessing) {
+        waitForTx(provider, txId, txHash)
```

Comment:
A transaction cannot be processing and relaying at the same time.

```suggestion
        waitForTx(provider, txId, txHash)
        continue
```